### PR TITLE
print-stack

### PR DESCRIPTION
**Pull request by Jim8y**
Print the execution stack while running the unit test contract.

### CODE REVIEW

**Comment by Jim8y** [comment]
@shargon

**Comment by shargon** [comment]
@Jim8y Didn't we already merge a PR that did this?

**Comment by Jim8y** [comment]
> @Jim8y Didn't we already merge a PR that did this?

Its in the core, but roman and chris think it is not good to add test features into the core, so i have to add them seperately into the test.

File: tests/Neo.SmartContract.TestEngine/Extensions.cs [hunk 1]
```diff
@@ -27,5 +31,45 @@ public static void DeployNativeContracts(this DataCache snapshot, Block persisti
...
+            {
+                value = Utility.StrictUTF8.GetString(byteArray);
+                return true;
+            }
+            catch (DecoderFallbackException)
+            {
+                value = default;
+                return false;
+            }
+            catch (ArgumentException)
```
Comment (shargon) [inline tests/Neo.SmartContract.TestEngine/Extensions.cs:47]: Catch all of them?

```suggestion
            catch
```

Comment (Jim8y) [reply tests/Neo.SmartContract.TestEngine/Extensions.cs:47]: not really necessary, but wont hurt.

**Review by shargon — COMMENTED** [review]
Wait for https://github.com/neo-project/neo/pull/3110 and use this extension here

**Comment by Jim8y** [comment]
> Wait for [neo-project/neo#3110](https://github.com/neo-project/neo/pull/3110) and use this extension here

Its in the unit tests, we can not access them here.

**Comment by shargon** [comment]
> > Wait for [neo-project/neo#3110](https://github.com/neo-project/neo/pull/3110) and use this extension here
> 
> Its in the unit tests, we can not access them here.

Is right, and is not better to reuse, and don't move the method outside vm? (https://github.com/neo-project/neo/pull/3110/files#diff-30362cba3801f54b23b08a98f1a0887bee8e3b611fd6cc8e412ec58800633f62L39)

**Comment by Jim8y** [comment]
> > > Wait for [neo-project/neo#3110](https://github.com/neo-project/neo/pull/3110) and use this extension here
> > 
> > 
> > Its in the unit tests, we can not access them here.
> 
> Is right, and is not better to reuse, and don't move the method outside vm? (https://github.com/neo-project/neo/pull/3110/files#diff-30362cba3801f54b23b08a98f1a0887bee8e3b611fd6cc8e412ec58800633f62L39)

That one is the one you merged, not in any release yet, for the same purpose here but have a bug in parsing string.

**Review by cschuchardt88 — APPROVED** [review]

**Comment by Jim8y** [comment]
@shargon good to merge now.

**Review by shargon — APPROVED** [review]

**Pull request merged**